File: 1_Baidu/1009/ans.java
import java.util.Scanner;
public class Main {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String str = in.next();
        in.close();
        
        int curr = 0;
        int anchor = 0;
        
        for(curr=0;curr<str.length();curr++) {
            if(curr+1 == str.length()||str.charAt(curr+1)!=str.charAt(anchor)) {
                System.out.print(str.charAt(anchor));
                if(curr > anchor) {
                    if(curr-anchor+1 ==2)System.out.print(str.charAt(curr));
                    if(curr-anchor+1 >2)System.out.print(String.valueOf(curr-anchor+1));
                }
                anchor = curr +1;
            };
        }
        
        System.out.println();
        
    }

}

File: 1_Baidu/1009/ans.cpp
#include <iostream>
using namespace  std;



int main() {
    string str;
    cin >> str;
    int curr;
    int anchor = 0;

    for (curr = 0;curr<str.size();curr++){
        if(curr+1==str.size()|| str[curr+1]!=str[curr]){
            cout <<  str[anchor];
            if(curr > anchor){
                if(curr -anchor+1 ==2) cout <<  str[curr];
                if(curr -anchor+1 >2){
                    for(char c :(""+to_string(curr-anchor+1))){
                        cout << c;
                    }
                }
            }
            anchor = curr+1;
        }
    }



    cout << endl;
    return 0;
}